.github/workflows/tests.yml

```diff
@@ -11,32 +11,48 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ["3.6", "3.7", "3.8", "3.9"]
+    defaults:
+      run:
+        shell: bash
     env:
       OS: ${{ matrix.os }}
       PYTHON: ${{ matrix.python-version }}
     name: Python ${{ matrix.python-version }} on OS ${{ matrix.os }}
     steps:
     - name: Acquire sources
-      uses: actions/checkout@v2
+      uses: actions/checkout@v2.3.4
+
     - name: Setup Python
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
         architecture: x64
-    - name: Setup Poetry
-      uses: Gr1N/setup-poetry@v4
-    - name: Configure poetry
-      # FROM https://github.com/northtree/poetry-github-actions/actions/runs/170795796/workflow
-      run: |
-        poetry config virtualenvs.in-project false
-        poetry config cache-dir ~/.poetry
-        poetry config virtualenvs.path ~/.poetry/venv
+
+    - name: Install and configure Poetry
+      uses: snok/install-poetry@v1.1.1
+      with:
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+        virtualenvs-path: .venv
```

Comment:
Just a remark: `virtualenv-path` is not set within the canonical example for Windows at https://github.com/snok/install-poetry#windows.